rust/lance-encoding/src/encodings/physical/fixed_size_binary.rs

```diff
@@ -0,0 +1,388 @@
+                for i in 0..arr.len() {
+                    let start = arr.offsets()[i] as usize;
+                    let end = arr.offsets()[i + 1] as usize;
+                    if start == end {
+                        // Null value, add byte_width bytes of zeroes
+                        values_vec.extend(std::iter::repeat(0).take(byte_width))
+                    } else {
+                        values_vec.extend_from_slice(&arr.values()[start..end]);
+                    }
+                }
+
+                values_vec
+            } else {
+                panic!("Array is not a string / binary array");
+            };
+
+            let len_buffer = values_buffer.len();
+            let values = ScalarBuffer::new(Buffer::from(values_buffer), 0, len_buffer);
+            Arc::new(UInt8Array::new(values, None)) as ArrayRef
+        })
+        .collect()
+}
+
+impl ArrayEncoder for FixedSizeBinaryEncoder {
+    fn encode(&self, arrays: &[ArrayRef], buffer_index: &mut u32) -> Result<EncodedArray> {
+        let byte_arrays = get_bytes_from_binary_arrays(arrays, self.byte_width);
+        let encoded_bytes = self.bytes_encoder.encode(&byte_arrays, buffer_index)?;
+
+        Ok(EncodedArray {
+            buffers: encoded_bytes.buffers,
+            encoding: pb::ArrayEncoding {
+                array_encoding: Some(pb::array_encoding::ArrayEncoding::FixedSizeBinary(
+                    Box::new(pb::FixedSizeBinary {
+                        bytes: Some(Box::new(encoded_bytes.encoding)),
+                        byte_width: self.byte_width as u32,
+                    }),
+                )),
+            },
+        })
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use std::{collections::HashMap, sync::Arc};
+
+    use arrow::array::LargeStringBuilder;
+    use arrow_array::{ArrayRef, LargeStringArray, StringArray};
+    use arrow_schema::{DataType, Field};
+
+    use crate::testing::{
+        check_round_trip_encoding_of_data, check_round_trip_encoding_random, TestCases,
+    };
+
+    #[test_log::test(tokio::test)]
+    async fn test_fixed_size_utf8_binary() {
+        let field = Field::new("", DataType::Utf8, false);
+        // This test only generates fixed size binary arrays anyway
+        check_round_trip_encoding_random(field, HashMap::new()).await;
+    }
+
+    #[test_log::test(tokio::test)]
+    async fn test_fixed_size_binary() {
+        let field = Field::new("", DataType::Binary, false);
+        check_round_trip_encoding_random(field, HashMap::new()).await;
+    }
+
+    #[test_log::test(tokio::test)]
+    async fn test_fixed_size_large_binary() {
+        let field = Field::new("", DataType::LargeBinary, true);
+        check_round_trip_encoding_random(field, HashMap::new()).await;
+    }
+
+    #[test_log::test(tokio::test)]
+    async fn test_fixed_size_large_utf8() {
+        let field = Field::new("", DataType::LargeUtf8, true);
+        check_round_trip_encoding_random(field, HashMap::new()).await;
+    }
```

Comment:
How are you generating fixed size arrays here?  Aren't these tests just using the normal binary encoding?

Reply:
I think this test actually generates fixed size arrays internally...check `rand_utf8()` in `generator.rs`
I also printed which encoding is used to double check. We should probably update the test, actually (maybe in a separate PR)